### PR TITLE
example(routing): adding routing example and e2e tests

### LINKED ISSUE #2650: example(router): add simple example app
@IgorMinar is going to take over this branch @matsko started to add a simple app + E2E tests with the router.

* https://github.com/matsko/angular/tree/routing_tests

### PR DESCRIPTION
**Pull request by matsko**
Closes #2650

### CODE REVIEW

File: modules/angular2/src/router/hash_location_strategy.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export class HashLocationStrategy extends LocationStrategy {
 
   getBaseHref(): string { return ''; }
 
-  path(): string { return this._location.hash; }
+  path(): string { return this._location.hash.substr(1); }
```
Comment (btford) [inline modules/angular2/src/router/hash_location_strategy.ts]: can you break this into its own commit for visibility?

**Comment by btford** [comment]
Other than my one comment LGTM. Split out that commit, then merge when CI is green :+1:

**Comment by matsko** [comment]
This is superseded by https://github.com/angular/angular/pull/2966
